assets/js/base/context/cart-checkout/shipping/index.js

```diff
@@ -165,8 +165,10 @@ export const ShippingDataProvider = ( { children } ) => {
 
 	// dispatch checkout error if there's a shipping error.
 	useEffect( () => {
-		dispatchActions.setHasError( currentErrorStatus.hasError );
-	}, [ currentErrorStatus, dispatchActions.setHasError ] );
+		if ( currentErrorStatus.hasError ) {
+			dispatchActions.setHasError( true );
+		}
+	}, [ currentErrorStatus.hasError, dispatchActions.setHasError ] );
```

Comment:
A problem with this change is it means if the shipping error is cleared, then the checkout error status won't get cleared. If there is potential for race conditions here, we may have to (not in this pull) add similar treatment for checkout hasError as what I did for the `isCalculating` state (keep count of things causing an error and increment/decrement count on those individual error state changes).

Reply:
Yeah... makes sense. I removed the conditional for now (dc1901580e7812b1eef6585b32d3e5f19ac21a62) and created a follow-up, so we don't block this PR: #2096. I can start working on that one once this PR is merged.

I just wonder what's the benefit on having an error counter with increment/decrement instead of an array of error ids (or even better, a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)). Both solutions would work, but a set of error ids would make our lives easier when debugging IMO. What do you think?

Reply:
> Both solutions would work, but a set of error ids would make our lives easier when debugging IMO. What do you think?

Good thoughts, depending on what you end up storing as references for tracking, you may want to use a WeakSet to protect against potential memory leaks (from unmounts etc). The only reason I suggested an error counter is because it's trivial to implement and memory impact is insignificant.